static/js/compose.js

```diff
@@ -500,6 +538,12 @@ function validate_stream_message_mentions(stream_id) {
             $("#sending-indicator").hide();
             return false;
         }
+        if (!exports.wildcard_mention_allowed()) {
+            compose_error(
+                i18n.t("You do not have permission to use wildcard mentions in this stream."),
+            );
+            return false;
+        }
```

Comment:
I think this check would be better as the first one in this block -- it lets us avoid calling `wildcard_mention_allowed` twice.  I took care of moving this.